network/src/peer_manager/mod.rs

```diff
@@ -459,31 +459,34 @@ where
         self.sample_connected_peers();
         match event {
             TransportNotification::NewConnection(mut conn) => {
+                let trusted_peers = self.trusted_peers.read().clone();
                 // TODO: This is right now a hack around having to feed trusted peers deeper in the outbound path.  Inbound ones are assigned at Noise handshake time.
                 if conn.metadata.origin == ConnectionOrigin::Outbound {
-                    let peer_role = self
-                        .trusted_peers
-                        .read()
+                    let peer_role = trusted_peers
                         .get(&conn.metadata.remote_peer_id)
                         .map_or(PeerRole::Unknown, |auth_context| auth_context.role);
                     conn.metadata.role = peer_role;
                 };
 
```

Comment:
seems like the rest of this code is only relevant if the peer is unknown?

also one could argue there's an asymmetry in your logic, what if I explicitly want to disallow someone by putting them as unknown or untrusted? they would still be slotted for a connection but have a degraded peer role.

Reply:
If your concern is that this will accept `Unknown` peers, I can add a check agains that rather than just checking if it doesn't exist in the `TrustedPeers` set.

This is where this `Unknown` in the `TrustedPeers` becomes a real pain from a naming perspective as well as determining what these peers are.


As far as the code here, the code below is relevant if the peer is `Unknown` and `Inbound`.  I'm not sure I want to support forcing nodes to be ignored off the bat.  It's a little confusing because the piece of code you're pointing at is about outbound peers, which (at this point) are only assigned roles here.

Reply:
Ah, I now see what you're talking about changing in discovery from one to the other.

We don't have a method of discovery that handles this asymmetry, as roles are assigned at creation time, and cannot be overridden at this point in connectivity manager.  I took this simplistic approach to ensure that seed roles override discovery roles, but in the future when more discovery approaches exist, we could make it such that roles are overridden over time.